packages/cli/src/controllers/invitation.controller.ts

```diff
@@ -0,0 +1,301 @@
+import { PostHogClient } from '@/posthog';
+
+@Service()
+@RestController('/invitations')
+export class InvitationController {
+	constructor(
+		private readonly config: Config,
+		private readonly logger: Logger,
+		private readonly internalHooks: IInternalHooksClass,
+		private readonly externalHooks: IExternalHooksClass,
+		private readonly userService: UserService,
+		private readonly roleService: RoleService,
+		private readonly mailer: UserManagementMailer,
+		private readonly postHog?: PostHogClient,
+	) {}
+
+	/**
+	 * Send email invite(s) to one or multiple users and create user shell(s).
+	 */
+
+	@Authorized(['global', 'owner'])
+	@Post('/')
+	async inviteUser(req: UserRequest.Invite) {
+		const isWithinUsersLimit = Container.get(License).isWithinUsersLimit();
+
+		if (isSamlLicensedAndEnabled()) {
+			this.logger.debug(
+				'SAML is enabled, so users are managed by the Identity Provider and cannot be added through invites',
+			);
+			throw new BadRequestError(
+				'SAML is enabled, so users are managed by the Identity Provider and cannot be added through invites',
+			);
+		}
+
+		if (!isWithinUsersLimit) {
+			this.logger.debug(
+				'Request to send email invite(s) to user(s) failed because the user limit quota has been reached',
+			);
+			throw new UnauthorizedError(RESPONSE_ERROR_MESSAGES.USERS_QUOTA_REACHED);
+		}
+
+		if (!this.config.getEnv('userManagement.isInstanceOwnerSetUp')) {
+			this.logger.debug(
+				'Request to send email invite(s) to user(s) failed because the owner account is not set up',
+			);
+			throw new BadRequestError('You must set up your own account before inviting others');
+		}
+
+		if (!Array.isArray(req.body)) {
+			this.logger.debug(
+				'Request to send email invite(s) to user(s) failed because the payload is not an array',
+				{
+					payload: req.body,
+				},
+			);
+			throw new BadRequestError('Invalid payload');
+		}
+
+		if (!req.body.length) return [];
+
+		const createUsers: { [key: string]: string | null } = {};
+		// Validate payload
+		req.body.forEach((invite) => {
+			if (typeof invite !== 'object' || !invite.email) {
+				throw new BadRequestError(
+					'Request to send email invite(s) to user(s) failed because the payload is not an array shaped Array<{ email: string }>',
+				);
+			}
+
+			if (!validator.isEmail(invite.email)) {
+				this.logger.debug('Invalid email in payload', { invalidEmail: invite.email });
+				throw new BadRequestError(
+					`Request to send email invite(s) to user(s) failed because of an invalid email address: ${invite.email}`,
+				);
+			}
+			createUsers[invite.email.toLowerCase()] = null;
+		});
+
+		const role = await this.roleService.findGlobalMemberRole();
+
+		if (!role) {
+			this.logger.error(
+				'Request to send email invite(s) to user(s) failed because no global member role was found in database',
+			);
+			throw new InternalServerError('Members role not found in database - inconsistent state');
+		}
+
```

Comment:
Can we please create a new `InvitationService` and move the code after this line into a method `inviteUsers` in that service? 
Then we can use that service directly in external hooks.